### PR TITLE
chore: use poetry as dependency manager

### PR DESCRIPTION
**Pull request by jorgebodega**

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -30,14 +29,18 @@ jobs:
           --health-retries 5
     steps:
       - uses: actions/checkout@v3
+      - uses: abatilo/actions-poetry@v2.1.5
+        with:
+          poetry-version: 1.2.0
+      - uses: actions/setup-python@v4
+        with:
+          python-version-file: .python-version
+          cache: 'poetry'
```
Comment (mercuriete) [inline .github/workflows/pull_request.yaml]: setup poetry in the same way you will do in the container

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -9,7 +9,6 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
-    container: "python:3.10-slim"
```
Comment (mercuriete) [inline .github/workflows/pull_request.yaml]: dont drop this line

File: .python-version [hunk 3]
```diff
@@ -0,0 +1 @@
+3.10.6
```
Comment (mercuriete) [inline .python-version]: enter at the end of the file

**Comment by mercuriete** [comment]
I recommend you to solve the problem of installing poetry using a Dockerfile on local.

File: .python-version (hunk 3 above)
Comment (jorgebodega) [reply .python-version]:
```suggestion
3.10.6

```

File: .github/workflows/pull_request.yaml (hunk 2 above)
Comment (jorgebodega) [reply .github/workflows/pull_request.yaml]: This line is the one in conflict with 

```yml
- uses: actions/setup-python@v4
        with:
          python-version-file: .python-version
          cache: 'poetry'
```

Just for testing right now

File: .github/workflows/pull_request.yaml (hunk 1 above)
Comment (jorgebodega) [reply .github/workflows/pull_request.yaml]: What? Don't understand

File: .github/workflows/pull_request.yaml [hunk 4]
```diff
@@ -28,14 +27,18 @@ jobs:
           --health-retries 5
     steps:
       - uses: actions/checkout@v3
+      - uses: abatilo/actions-poetry@v2.1.5
+        with:
+          poetry-version: 1.1.15
```
Comment (jorgebodega) [inline .github/workflows/pull_request.yaml]:
```suggestion
      - run: pip install poetry==1.1.15
```

**Comment by mercuriete** [comment]
did you check .github/workflow/main.yaml ?

**Comment by jorgebodega** [comment]
It should work as well.

`run: heroku run flask db upgrade -a ${{ secrets.HEROKU_APP_NAME }}` is the only line that could fail, but packages are installed as system dependencies, so it shouldn't

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+[tool.poetry]
+name = "oipie-api-core"
+version = "0.0.1"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (morenocantoj) [inline pyproject.toml]: Can we put some of our project examples here?
```toml
description="API to list recipes"
authors=["Jorge Bodega <jorge.bodega@acidtango.com", "Jaume Moreno <jaume@acidtango.com"]
```

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+license = "MIT"
+
+[tool.poetry.dependencies]
+dependency-injector = "4.40.0"
+Flask = "2.2.2"
+Flask-Migrate = "3.1.0"
+Flask-SQLAlchemy = "2.5.1"
+gunicorn = "20.1.0"
+psycopg2-binary = "2.9.3"
```
Comment (morenocantoj) [inline pyproject.toml:19]: I think we only need `psycopg2-binary` because `SQLAlchemy` needs it

**Review by morenocantoj — DISMISSED** [review]

**Comment by morenocantoj** [comment]
Also we should update `README.md` to reflect these changes in the dependency management

File: pyproject.toml (hunk 5 above)
Comment (jorgebodega) [reply pyproject.toml]: We can. Let me check.

File: pyproject.toml (hunk 6 above)
Comment (jorgebodega) [reply pyproject.toml:19]: That package is some kind of "peer" dependency, because SQLAlchemy does not install it.

**Review by mercuriete — DISMISSED** [review]
Remember to click on "Squash and merge"
20 commits are too much for 9 changes.

**Review by mercuriete — DISMISSED** [review]
LGTM!

**Review by mercuriete — APPROVED** [review]
LGTM!

**Pull request merged**